lib/awful/rules.lua

```diff
@@ -277,15 +277,9 @@ function rules.execute(c, props, callbacks)
                     t = atag.find_by_name(props.screen, t)
                 end
 
-                if type(value) == "function" then
-                    value = value(c, props)
+                if t then
+                    c:tags{ t }
                 end
-
-                if value then
-                    c.screen = value.screen
-                    c:tags{ value }
-                end
```

Comment:
This looks like a rebase gone wrong. The previous commit introduced all of this, so you should squash this into there.